File: module-domain/src/main/java/eu/geekhome/domain/configurable/StringFieldBuilder.java
package eu.geekhome.domain.configurable;

public class StringFieldBuilder implements FieldBuilder<String> {
    @Override
    public String fromPersistableString(String value) {
        return value;
    }

    @Override
    public String toPersistableString(String value) {
        return value;
    }
}

File: modules/domain/src/main/java/eu/automateeverything/domain/configurable/StringFieldBuilder.kt
package eu.automateeverything.domain.configurable

class StringFieldBuilder : FieldBuilder<String> {
    override fun fromPersistableString(value: String?): String {
        if (value == null) {
            return ""
        }

        return value
    }

    override fun toPersistableString(value: String): String {
        return value
    }
}